### PR TITLE
Added challenge to 01-numpy.md regarding slicing a vector vs an array

### PR DESCRIPTION
**Pull request by metavee**
Illustrates one of the differences between NumPy and MATLAB/intuition. Depending on the way you slice something, you may get back a 1D (N,) or 2D (Nx1) array, which can cause weird errors if you aren't expecting it.

### CODE REVIEW

File: 01-numpy.md [hunk 1]
```diff
@@ -703,3 +703,22 @@ the graphs will actually be squeezed together more closely.)
 > ## Moving plots around {.challenge}
 >
 > Modify the program to display the three plots on top of one another instead of side by side.
+
+> ## Converting between an array and a vector {.challenge}
+>
+> In some cases in NumPy, an vector of size N behaves differently than an Nx1 array.
```
Comment (QuLogic) [inline 01-numpy.md]: I don't think this challenge really explains _how_ it might behave differently, or why that might even be a problem?

Comment (metavee) [inline 01-numpy.md]: Fair point. In my experience, it generally results in strange-looking broadcasting errors.

Perhaps I could reformat this as a small code fragment that results in such an error, and the challenge would be to fix the error. Do you think that would work better?

**Comment by metavee** [comment]
I updated the challenge to try and guide the participants to the error in an example.

I'm not totally satisfied with how it turned out, but I found it quite challenging to come up with a clear and simple example. It would be great to get feedback on whether or not this is salvageable.

**Comment by valentina-s** [comment]
That looks good! Thank you for the contribution!

**Pull request merged**